compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -737,13 +737,35 @@ object Trees {
     type ThisTree[+T <: Untyped] = Splice[T]
   }
 
+  /** A tree representing a quote pattern `'{ type binding1; ...; body }` or `'[ type binding1; ...; body ]`.
+   *  `QuotePattern`s are created the type checker when typing an `untpd.Quote` in a pattern context.
+   *
+   *  `QuotePattern`s are checked are encoded into `unapply`s  in the `staging` phase.
+   *
+   *   The `bindings` contain the list of quote pattern type variable definitions (`Bind`s) in the oreder in
+   *   which they are defined in the source.
+   *
+   *   @param  bindings  Type variable definitions (`Bind` tree)
+   *   @param  body      Quoted pattern (without type variable definitions)
+   *   @param  quotes    A reference to the given `Quotes` instance in scope
```

Comment:
Does this actually need to be stored in the tree or could we look for the given when we need it?

Reply:
In theory, we could look for it again. But I preferred to keep track of it to ensure we do not change the instance. Maybe a macro-generated quote pattern could go wrong. I need to analyze this in depth if we want to remove this instance. Would be nicer though.